File: python/QuickSort.py
def partition(a, l, h):
    pvt = a[h]
    j = l - 1
    
    for k in range(l, h):
        if a[k] < pvt:
            j += 1
            swap(a, j, k)
    
    swap(a, j + 1, h)
    return j + 1

def swap(a, j, k):
    a[j], a[k] = a[k], a[j]

def qcksort(a, l, h):
    if l < h:
        pi = partition(a, l, h)
        qcksort(a, l, pi - 1)
        qcksort(a, pi + 1, h)
        
if __name__ == "__main__":
    a = [10, 7, 8, 9, 1, 5]
    size = len(a)
    print("El arreglo antes de ordenarlo: \n")
    for v in  a:
        print (v, end = ' ')
    print()
    qcksort(a, 0, size - 1)
    print("El arreglo despues de ordenarlo\n")
    for v in a:
        print (v, end = ' ')
    
    
    

File: javascript/QuickSort.js
function partition(a, l, h){
    let pvt = a[h];
    let j = l - 1;

    for (let k = l; k < h; k++){
        if (a[k] < pvt){
            j += 1;
            swap(a, j, k);
        }
    }
    swap(a, j + 1, h);
    return j + 1;
}

function swap(a, j, k){
    [a[j], a[k]] = [a[k], a[j]];
}

function qcksort(a, l, h){
    if (l < h){
        let pi = partition(a, l, h);
        qcksort(a, l, pi - 1);
        qcksort(a, pi + 1, h);
    }
}

let a = [10, 7, 8, 9, 1, 5];
let size = a.length;
console.log("El arreglo antes de ordenarlo: \n");
console.log(a);
console.log();
qcksort(a, 0, size - 1)
console.log("El arreglo despues de ordenarlo: \n");
console.log(a);